ports/wch/main.c

```diff
@@ -0,0 +1,54 @@
+#include "py/compile.h"
+#include "py/gc.h"
+#include "py/mperrno.h"
+#include "py/stackctrl.h"
+#include "shared/runtime/gchelper.h"
+#include "shared/runtime/pyexec.h"
+
+// Allocate memory for the MicroPython GC heap.
+static char heap[4096];
```

Comment:
If your linker hates you, heap[] may have an address like 0x20000123 - it's only promised to be byte aligned. (It'll probably be at least 4-byte aligned in practice...) Do you need to hedgee your bet and use __attribute__ aligned(16) or something to force it more strict so you can do floating point or other stuff in there that requires "better" alignment?